src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormatter.cs

```diff
@@ -125,18 +125,13 @@ public int GetCSharpIndentation(FormattingContext context, int projectedDocument
                 var baseProperty = result.GetType().GetProperty("BasePosition");
                 var basePosition = (int)baseProperty.GetValue(result);
                 var offsetProperty = result.GetType().GetProperty("Offset");
+
+                // IIndentationService always returns offset as the number of spaces.
                 var offset = (int)offsetProperty.GetValue(result);
 
                 var resultLine = changedText.Lines.GetLinePosition(basePosition);
                 var indentation = resultLine.Character + offset;
 
-                // IIndentationService always returns offset as the number of spaces.
-                // So if the client uses tabs instead of spaces, we need to convert accordingly.
-                if (!context.Options.InsertSpaces)
```

Comment:
I changed the place where I consumed this to operate on offset instead of levels. I needed to do that because the HTML formatter doesn't always format in exact multiples of levels. 
Example,
```html
<div class=foo
     id=fooid>
</div>
```